Formula/hwloc.rb

```diff
@@ -20,20 +20,37 @@ class Hwloc < Formula
 
   depends_on "pkg-config" => :build
   depends_on "cairo" => :optional
+  depends_on :x11 if build.with? "cairo" => ["with-x11"]
```

Comment:
I don't know if this is really needed, presumably this gets included transitively...

Reply:
Sorry, I misread, my apologies. That said I don't think it's worth adding this if it's included transitively. Additionally, `if build.with? "cairo"` doesn't work as expected (hence it's flagged by `brew audit --strict`)

Reply:
@MikeMcQuaid no worries, and thanks for maintaining such an awesome project!

I'm still confused, `brew audit --strict` did not flag any errors for me. Running `brew audit --strict hwloc` after installation gives me no output and 0 return code. My testing seemed to correctly handle the `if build.with? "cairo" => ["with-x11"]` too; the `--with-x` and `--without-x` were passed based on which `cairo` was installed. Maybe we have different rubocop versions?

Reply:
(To be clear, I care because the logic further down depends on `if build.with? "cairo" => ["with-x11"]` functioning correctly, which, in my extremely extensive testing it does. Combined with no issues flagged by `brew audit --strict hwloc` tested on two different machines, I'm inclined to leave this in.)

Reply:
It's actually a regular audit, not just --strict. The fact that it doesn't trigger here is a bug. Apparently, when we switched from using a regex to using a cop, the audit stopped working for the case where the right-hand-side is a multi-element array. That should be fixed in Homebrew/brew, but we'll enforce it via human cop until then.

As for this particular case, I think we may want to remove the X11 option from the `cairo` formula entirely since we don't want reverse dependencies of `cairo` ending up with different behavior depending on which options cairo was built with.

Reply:
https://github.com/Homebrew/homebrew-core/pull/27484 removes the X11 option from Cairo.